R/format-report.R

```diff
@@ -818,27 +818,35 @@ split_long_rows <- function(exported_func_df, n = 40) {
     # Expand each row if any of the splits are greater than 1
     dplyr::group_split() %>%
     purrr::map_dfr(function(row_data) {
-      n_chunks <- max(length(row_data$code_file_split[[1]]),
-                      length(row_data$documentation_split[[1]]),
-                      length(row_data$test_files_split[[1]]))
+      n_chunks <- max(
+        length(row_data$code_file_split[[1]]),
+        length(row_data$documentation_split[[1]]),
+        length(row_data$test_files_split[[1]]),
+        # Ensure the minimum value is at least 1
+        # i.e. when columns code_file, documentation, and test_files are all empty
+        1
+      )
 
       # Create a list of new rows
       purrr::map_dfr(1:n_chunks, function(i) {
         new_row <- row_data
 
+        code_split_len <- length(new_row$code_file_split[[1]])
+        doc_split_len <- length(new_row$documentation_split[[1]])
+        test_split_len <- length(new_row$test_files_split[[1]])
         # Extract split contents or default to an empty string
         new_row$code_file <- ifelse(
-          length(new_row$code_file_split[[1]]) >= i,
+          code_split_len >= i && code_split_len != 0,
```

Comment:
Hmm, `i` starts at 1 and goes through `n_chunks`.  So how could the second condition (`code_split_len != 0`) ever be false if the first condition (`code_split_len >= i`) is true? (Same applies to the conditions on lines 844 and 849.)

I don't see a detailed description of what was going wrong in the old code in either the PR description or commit message, but at a glance I think it was this:

 * in these cases `n_chunks` was 0
 * as a result, the map_dfr loop was over `c(1, 0)`

Now you're preventing that invalid sequence with the max call.  That's fine, but it's confusing to add a condition that's always true, even if you wanted to defensively guard against a regression.

My suggestion:

 * drop the second condition (unless I'm missing something and it can evaluate to false)
 * prefer `seq_len(n_chunks)` to `1:n_chunks`; in general if you're expecting a positive N `seq_len(N)` is good practice because it avoids the surprising case of 1:N going the other direction

Reply:
Yeah good catch; you're right I dont need those checks anymore. I originally fixed the issue (not erroring out) using those second conditions, though ultimately adding `1` to the `n_chunks` definition is what fixed it.

> in these cases n_chunks was 0...as a result, the map_dfr loop was over c(1, 0)

Correct. I added some more details to the latest commit as to why that was originally added